lib/pdk/tests/unit.rb

```diff
@@ -21,6 +22,36 @@ def self.invoke(tests, report = nil)
         output = PDK::CLI::Exec.execute(cmd(tests))
         report.write(output) if report
       end
+
+      # @return array of { :id, :full_description }
+      def self.list
+        PDK::Util::Bundler.ensure_bundle!
+        PDK::Util::Bundler.ensure_binstubs!('rspec-core')
+
+        command_argv = [File.join(PDK::Util.module_root, 'bin', 'rspec'), '--dry-run', '--format', 'json']
+        command_argv.unshift('ruby') if Gem.win_platform?
+        list_command = PDK::CLI::Exec::Command.new(*command_argv)
+        list_command.context = :module
+        output = list_command.execute!
+
+        rspec_json_output = JSON.parse(output[:stdout])
+        if rspec_json_output['examples'].empty?
+          rspec_message = rspec_json_output['messages'][0]
+          if rspec_message == 'No examples found.' # rubocop:disable Style/GuardClause
```

Comment:
You can get rid of this rubocop disable by rewriting this conditional as a suffix to the return statement and letting it fall through to the raise if it doesn't match

```
return [] if rspec_message == 'No examples found.'

raise PDK::CLI::FatalError, _(...)
```